man/man8/radcrypt.8

```diff
@@ -20,6 +23,16 @@ In this case \fIhashed_password\fP will be checked to see if it matches
 status will be 1, otherwise "Password BAD" will be printed and exit status
 will be 0 (Note this is the opposite of a normal successful shell status).
 
+That said, it should be noted that the current (August 2017) best practice is to 
+not use or store passwords at all. Rather, if possible, administrators are encouraged 
+to get rid of passwords and use X509 certs instead. Supplicant behaviour is poor on 
+macOS/iOS and Windows. If you use EAP-TTLS, EAP-PEAP etc... anyone sufficiently motivated 
+will be able to harvest your user's passwords.
+
+However, if generating high quality hashed passwords is a priority, dovecot's doveadm
+utility is quite useful (e.g. doveadm pw -s SSHA512) or openssl (echo -n "pw" | 
+openssl dgst -sha512).
+
```

Comment:
Current best practice is to use certificate-based authentication rather than relying on passwords. However, other external tools such as openssl can generate high quality hashed passwords if needed, e.g. "echo -n 'password' | openssl dgst -sha512".

_I would only mention openssl here as the server already has high dependency on that anyway, and it is also used to check the hashes, so it can generate them all. I'm sure there are plenty of other utilities that can generate them, so it doesn't really make much sense to me to list any more._